### PR TITLE
Select link correctness fix

### PR DESCRIPTION
**Pull request by Jake-Moss**
This is a potential fix for #442, with this change the output for select link run on link 9 is 
```
Assignment:
    link_id   a_ab   a_ba  a_tot
0         1  110.0    0.0  110.0
1         2   50.0    0.0   50.0
2         3  110.0   50.0  160.0
3         4    0.0    0.0    0.0
4         5    0.0    0.0    0.0
5         7  800.0   70.0  870.0
6         9  500.0   50.0  550.0
7        11   50.0  500.0  550.0
8        13   50.0  110.0  160.0
9        15    0.0    0.0    0.0
10       17  300.0   20.0  320.0
11       19  100.0  610.0  710.0
12       21  350.0  130.0  480.0
13       23  800.0   70.0  870.0

Select link
    link_id  class_a_trace_a_ab  class_a_trace_a_ba  class_a_trace_a_tot
0         1                 0.0                 0.0                  0.0
1         2                 0.0                 0.0                  0.0
2         3                 0.0                 0.0                  0.0
3         4                 0.0                 0.0                  0.0
4         5                 0.0                 0.0                  0.0
5         7               500.0                 0.0                500.0
6         9               500.0                 0.0                500.0
7        11                 0.0               500.0                500.0
8        13                 0.0                 0.0                  0.0
9        15                 0.0                 0.0                  0.0
10       17                 0.0                 0.0                  0.0
11       19                 0.0               500.0                500.0
12       21                 0.0                 0.0                  0.0
13       23               500.0                 0.0                500.0
```

The multithreading issue still remains but I have made progress on a refactor locally.

This is marked as draft as a real test needs to be added.

### CODE REVIEW

**Review by pedrocamargo — APPROVED** [review]
It does look like a quite obvious fix!!!

File: aequilibrae/paths/results/assignment_results.py [hunk 1]
```diff
@@ -109,6 +109,7 @@ def prepare(self, graph: Graph, matrix: AequilibraeMatrix) -> None:
         self.crosswalk = np.zeros(graph.graph.shape[0], self.__integer_type)
         self.crosswalk[graph.graph.__supernet_id__.values] = graph.graph.__compressed_id__.values
```
Comment (pedrocamargo) [inline aequilibrae/paths/results/assignment_results.py:110]: Do we use the crosswalk for anything?

Comment (Jake-Moss) [reply aequilibrae/paths/results/assignment_results.py:110]: Yeah, its still used within `all_or_nothing.py` and `linear_approximation.py`
```python
aequilibrae/paths/all_or_nothing.py
80:            self.results.link_loads, self.results.compact_link_loads, self.results.crosswalk, self.results.cores
aequilibrae/paths/linear_approximation.py
389:                aggregate_link_costs(cost, c.graph.compact_cost, c.results.crosswalk)
486:                    aggregate_link_costs(self.congested_time[:], c.graph.compact_skims[:, k], c.results.crosswalk)
aequilibrae/paths/results/assignment_results.py
 42:        self.crosswalk = np.array([])  # crosswalk between compact graph link IDs and actual link IDs
109:        self.crosswalk = np.zeros(graph.graph.shape[0], self.__integer_type)
110:        self.crosswalk[graph.graph.__supernet_id__.values] = graph.graph.__compressed_id__.values
```

**Comment by pedrocamargo** [comment]
Where are we on this, @Jake-Moss ?

**Comment by Jake-Moss** [comment]
I’m not sure what’s up with the documentation CI, but I’ll take a look at that tomorrow. Other than that I believe it’s good to go and solves both the correctness issue and the multithreading issues.

When I tested it on Chicago Sketch the results were comparable to the assignment. 

This pr also includes the testing data provided to us by Kai Tang as a test where it compares the assignment to the select link results with links chosen to cover all flow.

**Review by pedrocamargo — APPROVED** [review]

**Pull request merged**